lib/matplotlib/type1font.py

```diff
@@ -196,101 +410,93 @@ def _encrypt(plaintext, key, ndiscard=4):
-        prop = {'weight': 'Regular', 'ItalicAngle': 0.0, 'isFixedPitch': False,
+        prop = {'Weight': 'Regular', 'ItalicAngle': 0.0, 'isFixedPitch': False,
                 'UnderlinePosition': -100, 'UnderlineThickness': 50}
-        filtered = ((token, value)
-                    for token, value in self._tokens(self.parts[0])
-                    if token is not tok_whitespace)
+        pos = {}
         # The spec calls this an ASCII format; in Python 2.x we could
         # just treat the strings and names as opaque bytes but let's
         # turn them into proper Unicode, and be lenient in case of high bytes.
         def convert(x): return x.decode('ascii', 'replace')
-        for token, value in filtered:
-            if token is tok_name and value.startswith(b'/'):
-                key = convert(value[1:])
-                token, value = next(filtered)
-                if token is tok_name:
-                    if value in (b'true', b'false'):
-                        value = value == b'true'
-                    else:
-                        value = convert(value.lstrip(b'/'))
-                elif token is tok_string:
-                    value = convert(value.lstrip(b'(').rstrip(b')'))
-                elif token is tok_number:
-                    if b'.' in value:
-                        value = float(value)
-                    else:
-                        value = int(value)
-                else:  # more complicated value such as an array
-                    value = None
-                if key != 'FontInfo' and value is not None:
-                    prop[key] = value
+        data = self.parts[0] + self.decrypted
+
+        source = _tokenize(data, True)
+        while True:
+            # See if there is a key to be assigned a value
+            # e.g. /FontName in /FontName /Helvetica def
+            try:
+                token = next(source)
+            except StopIteration:
+                break
+            if token.is_delim():
+                # skip over this - we want top-level keys only
+                _expression(token, source, data)
+            if token.is_slash_name():
+                key = convert(token.value[1:])
+                keypos = token.pos
+            else:
+                continue
+
+            # Some values need special parsing
+            if key in ('Subrs', 'CharStrings', 'Encoding', 'OtherSubrs'):
+                prop[key], endpos = {
+                    'Subrs': self._parse_subrs,
+                    'CharStrings': self._parse_charstrings,
+                    'Encoding': self._parse_encoding,
+                    'OtherSubrs': self._parse_othersubrs
+                }[key](source, data)
+                pos.setdefault(key, []).append((keypos, endpos))
+                continue
+
+            # General case
+            try:
+                token = next(source)
+            except StopIteration:
+                break
+            if token.is_name(b'true', b'false'):
```

Comment:
perhaps(?) this switch could move to `_Token.parsed_value()` with appropriate overrides on each _Token subclass?  (An possibly also rename `_Token.value` to `_Token.raw` and then have the other be `_Token.parsed()` or whatnot.)